toponetx/utils/normalization.py

```diff
@@ -0,0 +1,359 @@
+"""Normalize of adjacency, boundary matrices of complexes."""
+
+import numpy as np
+import scipy.sparse.linalg as spl
+from numpy import ndarray
+from numpy.linalg import pinv
+from scipy.sparse import coo_matrix, diags, identity
+
+__all__ = [
+    "normalize_laplacian",
+    "normalize_x_laplacian",
+    "kipf_adjacency_matrix_normalization",
+    "asymmetric_xu_normalization",
+    "bunch_normalization",
+    "_compute_B1_normalized",
+    "_compute_B1T_normalized",
+    "_compute_B2_normalized",
+    "_compute_B2T_normalized",
+    "_compute_D1",
+    "_compute_D2",
+    "_compute_D3",
+    "_compute_D5",
+]
+
+
+def normalize_laplacian(L):
+    """Normalize the Laplacian matrix.
+
+    Parameters
+    ----------
+        L (scipy.sparse.csr_matrix): The Laplacian matrix.
+
+    Returns
+    -------
+        scipy.sparse.csr_matrix: The normalized Laplacian matrix.
+
+    Notes
+    -----
+        This function normalizes the Laplacian matrix by dividing it by the largest eigenvalue.
+    """
```

Comment:
The documentation style in this file does not follow NumPy's Conventions and will make problems with the documentation website. I'll illustrate how it should look like based on this example, can you please update it throughout the file?

```python3
"""Normalise the Laplacian matrix.

This function normalises the Laplacian matrix by dividing it by its largest eigenvalue.

Parameters
----------
L : scipy.sparse.scr_matrix
    The Laplacian matrix.

Returns
-------
scipy.sparse.csr_matrix
    The normalised Laplacian matrix.
"""

Reply:
Sure, one question, we should be consistent with the sparse type all over. Which type should we adapt? csr_matrix or coo?

Reply:
The output should be csr -- coo is really only for construction of matrices
See intended usage here
https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.coo_matrix.html